### PR TITLE
Use change directory instead of chdir on Windows

### PR DESCRIPTION
**Pull request by hughbe**
`chdir(1)` doesn't exist on Windows - use the `SetCurrentDirectory` Win32 API instead.

### CODE REVIEW

**Comment by ddunbar** [comment]
This code probably predates our use of LLVM, can we use one of LLVM's portable functions for this?

**Comment by hughbe** [comment]
I did look for this when I was writing this code. I googled around, searched the LLVM source code, and went on http://llvm.org/docs/doxygen/html/namespacellvm_1_1sys_1_1fs.html. Couldn't find anything though

**Comment by hughbe** [comment]
I've updated the PR to match your suggestion - I'm not extremely well versed in C++ best practices, but I think a static method in a class `PlatformUtility` is OK?

**Comment by hughbe** [comment]
Just realised I made a mistake by incorrectly inverting an if. Will fix in the next few hours

File: include/llbuild/Basic/PlatformUtility.h [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+// common functions.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLBUILD_BASIC_PLATFORMUTILITY_H
+#define LLBUILD_BASIC_PLATFORMUTILITY_H
+
+namespace llbuild {
+namespace basic {
+class PlatformUtility {
```
Comment (ddunbar) [inline include/llbuild/Basic/PlatformUtility.h]: I'd just make this a free function, and if you really want it to be in a namespace add another one inside llbuild::basic (e.g., `llbuild::basic::sys` would make sense to me).

File: lib/Basic/PlatformUtility.cpp [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+#include "llbuild/Basic/PlatformUtility.h"
+
+#if defined(_WIN32)
```
Comment (ddunbar) [inline lib/Basic/PlatformUtility.cpp]: Can we define WIN32_LEAN_AND_MEAN here, and maybe have a local Windows compatibility header which does that?

File: include/llbuild/Basic/PlatformUtility.h (hunk 1 above)
Comment (hughbe) [inline include/llbuild/Basic/PlatformUtility.h]: I added it into `llbuild::basic::sys` for a bit of structure

File: lib/Basic/PlatformUtility.cpp (hunk 2 above)
Comment (hughbe) [inline lib/Basic/PlatformUtility.cpp]: Added `LeanWindows.h`

File: include/llbuild/Basic/LeanWindows.h [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+
+#ifndef LLBUILD_BASIC_SHELLUTILITY_H
+#define LLBUILD_BASIC_SHELLUTILITY_H
+
+#if !defined(_WIN32)
+#error "LeanWindows.h should only be included on Windows."
+#endif
+
+#define NOMINMAX
```
Comment (ddunbar) [inline include/llbuild/Basic/LeanWindows.h]: Hah, didn't no about this one. Is there also NONEARFAR? :)

**Comment by ddunbar** [comment]
@swift-ci please test

File: include/llbuild/Basic/LeanWindows.h (hunk 3 above)
Comment (hughbe) [inline include/llbuild/Basic/LeanWindows.h]: Doesn't seem like it, as this wonderfully worded article complains about - http://lolengine.net/blog/2011/3/4/fuck-you-microsoft-near-far-macros

Comment (ddunbar) [inline include/llbuild/Basic/LeanWindows.h]: LOL :)

**Comment by hughbe** [comment]
I realised I made a typo in `LeanWindows.h` by not changing the `#ifndef` at the top of the file to the new file after copy-pasting from another header.
While I was at it, I moved `LeanWindows.h` into lib/Basic

**Comment by ddunbar** [comment]
@swift-ci please test

**Comment by ddunbar** [comment]
Failures appear unrelated.

@swift-ci please test

**Pull request merged**